src/core/ext/xds/xds_client.cc

```diff
@@ -522,27 +518,53 @@ void XdsClient::ChannelState::UnsubscribeLocked(const XdsResourceType* type,
   }
 }
 
-void XdsClient::ChannelState::OnConnectivityStateChange(absl::Status status) {
+void XdsClient::ChannelState::OnConnectivityFailure(absl::Status status) {
   {
     MutexLock lock(&xds_client_->mu_);
-    OnConnectivityStateChangeLocked(std::move(status));
+    SetChannelStatusLocked(std::move(status));
   }
   xds_client_->work_serializer_.DrainQueue();
 }
 
-void XdsClient::ChannelState::OnConnectivityStateChangeLocked(
-    absl::Status status) {
-  if (!shutting_down_) {
-    // Notify all watchers of error.
-    gpr_log(GPR_INFO,
-            "[xds_client %p] xds channel for server %s in "
-            "state TRANSIENT_FAILURE: %s",
-            xds_client(), server_.server_uri.c_str(),
-            status.ToString().c_str());
-    xds_client_->NotifyOnErrorLocked(absl::UnavailableError(
-        absl::StrCat("xds channel in TRANSIENT_FAILURE, connectivity error: ",
-                     status.ToString())));
+void XdsClient::ChannelState::SetChannelStatusLocked(absl::Status status) {
```

Comment:
Might be a dumb question - Why isn't there a single queue for the error notifications and actual updates?

Reply:
There is.  All of the notifications go through the same `WorkSerializer`.